lib/miq_automation_engine/engine/miq_ae_service/miq_ae_service_rbac.rb

```diff
@@ -0,0 +1,53 @@
+module MiqAeMethodService
+  module MiqAeServiceRbac
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def find_ar_object_by_id(id)
+        if rbac_enabled?
+          Rbac.filtered(model.where(:id => id), :user => workspace.ae_user).first
+        else
+          model.find(*id)
+        end
+      end
+
+      def all
+        objs = rbac_enabled? ? Rbac.filtered(model, :user => workspace.ae_user) : model.all
+        wrap_results(objs)
+      end
+
+      def count
+        rbac_enabled? ? Rbac.filtered(model, :user => workspace.ae_user).count : model.count
+      end
+
+      def first
+        all.first
```

Comment:
@gmcculloug 
Is this is for the first method or the all method? I think we need the limit for the first method.